runtime-light/component/component.cpp

```diff
@@ -48,8 +51,30 @@ void ComponentState::init_script_execution() noexcept {
 template<ComponentKind kind>
 task_t<void> ComponentState::run_component_prologue() noexcept {
   static_assert(kind != ComponentKind::Invalid);
-
   component_kind_ = kind;
+
+  // common initialization
+  auto &superglobals{php_script_mutable_globals_singleton.get_superglobals()};
```

Comment:
Maybe move it to separate function like `common_superglobals_initialization`. Because in the future, body of `run_component_prologue` may become too complex.